### ISSUE TITLE
yamllint interprets indentation as whitespace.

### ISSUE THREAD

**Issue by damngamerz** [adrienverge/yamllint#39]
YAML files such as configuration for travis have nested settings requiring indentation. yamllint currently doesn't know about any such scenarios.

Steps to reproduce

1. File abc.yml
```
---
language: ruby
rvm:
- 2.2.5

install: gem install jekyll

script: jekyll build

env:
  global:
  - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
sudo: false
```
2. While using yamllint `yamllint abc.yml`
```
yamllint abc.yml
abc.yml
  11:3      error    wrong indentation: expected 0 but found 2  (indentation)
```
But as we can see `global` is a setting for `env`, and it should be indented.
we can also see that yamllint does mix up nested blocks with the indentation style with `-`. As we look in the above configuration, the `- xxx` items needs all indentation to pass the check.Any thoughts about this behaviour?

For more info follow https://github.com/coala/coala-bears/issues/1252.

Additional Info
yamllint version 1.6.0

**Comment by adrienverge** [comment]
Hi,

I you use the default conf, the reported error is normal. In the [`default` configuration](http://yamllint.readthedocs.io/en/latest/configuration.html#default-configuration), sequences *must* be indented and you should write:

```yaml
env:
  global:
    - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
```

If you don't want to indent sequences, you can either use the `relaxed` rule:

```shell
yamllint -d relaxed abc.yaml
```

... write you own conf:

```shell
# file: .yamllint

extends: default
rules:
  indentation:
    indent-sequences: false
```

... or directly:

```shell
yamllint -d "{extends: default, rules: {indentation: {indent-sequences: false}}}" abc.yaml
```

---

However, the fact that:

```yaml
rvm:
- 2.2.5
```

is not detected as a problem seems to be a bug... I'll look into it as soon as I have time.

**Comment by damngamerz** [comment]
@adrienverge Hey,
In `default` configuration 
indentation:
indent-sequences is consistent 
I think there's some serious bug with it 
So,in my original `abc.yml`
It should give error at line no 4 and 12.
```
rvm:
- 2.2.5
```
and
```
global:
- NOKOGIRI_USE_SYSTEM_LIBRARIES=true
```
Now,
I tried as you have suggested
```
env:
  global:
    - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
```
Now using default config with `abc.yml` should pass and give error at
```
rvm:
- 2.2.5
```
As you have pointed it out.Instead its giving me
```
yamllint abc.yml 
abc.yml
  11:3      error    wrong indentation: expected 0 but found 2  (indentation)
  12:5      error    wrong indentation: expected 2 but found 4  (indentation)
```
Hope this info helps in fixing this bug.
Btw tests successfully passes with `abc.yml` as
```
---
language: ruby
rvm:
  - 2.2.5

install: gem install jekyll

script: jekyll build

env:
  global:
    - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
sudo: false
```
Thanks for pointing that out :+1:

**Comment by adrienverge** [comment]
> In default configuration
> indentation:
> indent-sequences is consistent

No, it's really:

```yaml
 indentation:
    indent-sequences: true  # not 'consistent'
```

(You can check it [in the source](https://github.com/adrienverge/yamllint/blob/master/yamllint/conf/default.yaml#L35) also.)

I think the bug can simply be narrowed to:

```yaml
a:
- b
```

not being detected with `indent-sequences: true`.

**Comment by adrienverge** [comment]
Hey @damngamerz, I just wrote [a fix](https://github.com/adrienverge/yamllint/commit/228c47ab770eb9981b64211edff846d053a6ad74) and released version 1.6.1.

Now it should give an error at:

```yaml
rvm:
- 2.2.5
```

... and not at line 11 anymore.

By the way, if you prefer not-indented-list style, I suggest you create a `.yamllint` conf file in your repo, this is pretty handy!

Hope this helps.

**Comment by damngamerz** [comment]
@adrienverge Thanks! that works perfectly :+1:
